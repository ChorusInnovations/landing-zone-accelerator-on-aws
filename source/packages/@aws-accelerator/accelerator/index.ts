/**
 *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

export * from './lib/accelerator';
export * from './lib/config-repository';
export * from './lib/pipeline';
export * from './lib/stacks/accounts-stack';
export * from './lib/stacks/default-stack';
export * from './lib/stacks/dependencies-stack';
export * from './lib/stacks/network-1-stack';
export * from './lib/stacks/network-2-stack';
export * from './lib/stacks/network-3-stack';
export * from './lib/stacks/operations-stack';
export * from './lib/stacks/organizations-stack';
export * from './lib/stacks/pipeline-stack';
export * from './lib/stacks/security-audit-stack';
export * from './lib/stacks/security-stack';
export * from './lib/stacks/validate-stack';
export * from './lib/toolkit';
