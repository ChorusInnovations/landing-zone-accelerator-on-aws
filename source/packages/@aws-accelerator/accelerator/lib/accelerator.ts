/**
 *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */
import { AccountsConfig, GlobalConfig, OrganizationConfig, SecurityConfig } from '@aws-accelerator/config';
import { throttlingBackOff } from '@aws-accelerator/utils';
import { AssumeProfilePlugin } from '@aws-cdk-extensions/cdk-plugin-assume-role';
import { DescribeOrganizationCommand, OrganizationsClient, paginateListAccounts } from '@aws-sdk/client-organizations';
import { RequireApproval } from 'aws-cdk/lib/diff';
import { PluginHost } from 'aws-cdk/lib/plugin';
import { AcceleratorToolkit } from './toolkit';

export enum AcceleratorStage {
  PIPELINE = 'pipeline',
  /**
   * Validate Stage - Verify the configuration files and environment
   */
  VALIDATE = 'validate',
  ORGANIZATIONS = 'organizations',
  LOGGING = 'logging',
  /**
   * Accounts Stage - Handle all Organization and Accounts actions
   */
  ACCOUNTS = 'accounts',
  DEPENDENCIES = 'dependencies',
  SECURITY = 'security',
  OPERATIONS = 'operations',
  NETWORKING = 'networking',
  'SECURITY-AUDIT' = 'security-audit',
}

/**
 *
 */
export interface AcceleratorProps {
  command: string;
  configDirPath: string;
  parallel: boolean;
  stage: string;
  account: string;
  region: string;
  partition: string;
  requireApproval: RequireApproval;
}

/**
 * Wrapper around the CdkToolkit. The Accelerator defines this wrapper to add
 * the following functionality:
 *
 * - x
 * - y
 * - z
 */
export abstract class Accelerator {
  static isSupportedStage(stage: AcceleratorStage): boolean {
    return Object.values(AcceleratorStage).includes(stage);
  }

  /**
   *
   * @returns
   */
  static async run(props: AcceleratorProps): Promise<void> {
    //
    // When an account and region is specified, execute as single stack
    //
    if (props.account || props.region) {
      if (props.account && props.region === undefined) {
        throw new Error(`Account set to ${props.account}, but region is undefined`);
      }
      if (props.region && props.account === undefined) {
        throw new Error(`Region set to ${props.region}, but region is undefined`);
      }
      return await AcceleratorToolkit.execute({
        command: props.command,
        accountId: props.account,
        region: props.region,
        stage: props.stage,
        configDirPath: props.configDirPath,
        requireApproval: props.requireApproval,
      });
    }

    //
    // Read in all Accelerator Configuration files here, then pass the objects
    // to the stacks that need them. Exceptions are thrown if any of the
    // configuration files are malformed.
    //
    const globalConfig = GlobalConfig.load(props.configDirPath);
    const organizationsConfig = OrganizationConfig.load(props.configDirPath);
    const accountsConfig = AccountsConfig.load(props.configDirPath);
    const securityConfig = SecurityConfig.load(props.configDirPath);

    //
    // Load Plugins
    //
    const assumeRolePlugin = new AssumeProfilePlugin({
      // TODO: Read this from arg
      assumeRoleName: organizationsConfig['organizations-access-role'],
      assumeRoleDuration: 3600,
    });
    assumeRolePlugin.init(PluginHost.instance);

    //
    // NOTE: We do some early environment validation here before we kick off the
    //       CodePipeline that has a built in validation-stack
    //

    //
    // Verify AWS Organizations has been enabled
    //
    const organizationsClient = new OrganizationsClient({});
    await throttlingBackOff(() =>
      organizationsClient.send(new DescribeOrganizationCommand({})).catch(error => {
        if (error.name === 'AWSOrganizationsNotInUseException') {
          throw new Error(error.message);
        }
        throw new Error(error);
      }),
    );

    //
    // Create a dictionary of all AWS Account IDs.
    //
    // TODO: Add functionality to read in a map from file for disconnected regions
    //
    const accountIds: { [name: string]: string } = {};
    for await (const page of paginateListAccounts({ client: organizationsClient }, {})) {
      for (const account of page.Accounts ?? []) {
        if (account.Email && account.Id) {
          accountIds[account.Email] = account.Id;
        }
      }
    }

    //
    // Execute Bootstrap stacks for all identified accounts
    //
    if (props.command == 'bootstrap') {
      const trustedAccountId = accountIds[accountsConfig['mandatory-accounts'].management.email];
      for (const region of globalConfig['enabled-regions']) {
        for (const account of Object.values(accountsConfig['mandatory-accounts'])) {
          const accountId = accountIds[account.email];
          await AcceleratorToolkit.execute({
            command: props.command,
            accountId,
            region,
            partition: props.partition,
            trustedAccountId,
            requireApproval: props.requireApproval,
          });
        }
        for (const account of Object.values(accountsConfig['workload-accounts'])) {
          const accountId = accountIds[account.email];
          await AcceleratorToolkit.execute({
            command: props.command,
            accountId,
            region,
            partition: props.partition,
            trustedAccountId,
            requireApproval: props.requireApproval,
          });
        }
      }
      return;
    }

    // TODO: Need to decide the mandatory accounts for an accelerator --
    // Control Tower: To start a well-planned OU structure in your landing zone, AWS Control Tower
    // sets up a Security OU for you. This OU contains three shared accounts: the management
    // (primary) account, the log archive account, and the security audit account (also referred to
    // as the audit account).

    switch (props.stage) {
      case AcceleratorStage.VALIDATE:
      case AcceleratorStage.ACCOUNTS:
        break;
      case AcceleratorStage.ORGANIZATIONS:
        const managementAccount = accountsConfig['mandatory-accounts'].management;
        for (const region of globalConfig['enabled-regions']) {
          await AcceleratorToolkit.execute({
            command: props.command,
            accountId: accountIds[managementAccount.email],
            region: region,
            stage: props.stage,
            configDirPath: props.configDirPath,
            requireApproval: props.requireApproval,
          });
        }
        break;

      //
      // Apply these stacks to all account / regions. The contents of these stacks are dynamically
      // built from the inputted configuration files during stack construction
      //
      case AcceleratorStage.LOGGING:
      case AcceleratorStage.DEPENDENCIES:
      case AcceleratorStage.SECURITY:
        for (const region of globalConfig['enabled-regions'].filter(
          item => securityConfig.getExcludeRegions()?.indexOf(item) === -1,
        )) {
          for (const account of Object.values(accountsConfig['mandatory-accounts'])) {
            console.log(`${region} to be implemented on for ${account.email} account`);
            const accountId = accountIds[account.email];
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId,
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
          for (const account of Object.values(accountsConfig['workload-accounts'])) {
            const accountId = accountIds[account.email];
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId,
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
        }
        break;
      case AcceleratorStage.OPERATIONS:
      case AcceleratorStage.NETWORKING:
        for (const region of globalConfig['enabled-regions']) {
          for (const account of Object.values(accountsConfig['mandatory-accounts'])) {
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId: accountIds[account.email],
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
          for (const account of Object.values(accountsConfig['workload-accounts'])) {
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId: accountIds[account.email],
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
        }
        break;
      case AcceleratorStage['SECURITY-AUDIT']:
        const auditAccountName = securityConfig.getDelegatedAccountName();
        if (accountsConfig.accountExists(auditAccountName)) {
          for (const region of globalConfig['enabled-regions'].filter(
            item => securityConfig.getExcludeRegions()?.indexOf(item) === -1,
          )) {
            const auditAccountEmail = accountsConfig.getEmail(auditAccountName);
            const accountId = accountIds[auditAccountEmail];
            await AcceleratorToolkit.execute({
              command: props.command,
              accountId,
              region,
              stage: props.stage,
              configDirPath: props.configDirPath,
              requireApproval: props.requireApproval,
            });
          }
        } else {
          throw new Error(`Security audit delegated account ${auditAccountName} not found.`);
        }
        break;
      default:
        throw new Error(`Unknown stage: ${props.stage}`);
    }
  }
}
