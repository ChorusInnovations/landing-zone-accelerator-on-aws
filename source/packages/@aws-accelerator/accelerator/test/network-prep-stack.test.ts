/**
 *  Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import {
  AccountsConfig,
  GlobalConfig,
  IamConfig,
  NetworkConfig,
  OrganizationConfig,
  SecurityConfig,
} from '@aws-accelerator/config';
import * as cdk from 'aws-cdk-lib';
import * as path from 'path';
import { AcceleratorStackNames } from '../lib/accelerator';
import { AcceleratorStage } from '../lib/accelerator-stage';
import { AcceleratorStackProps } from '../lib/stacks/accelerator-stack';
import { NetworkPrepStack } from '../lib/stacks/network-prep-stack';

const testNamePrefix = 'Construct(NetworkPrepStack): ';

/**
 * NetworkPrepStack
 */
const app = new cdk.App({
  context: { 'config-dir': path.join(__dirname, 'configs/all-enabled') },
});
const configDirPath = app.node.tryGetContext('config-dir');

const props: AcceleratorStackProps = {
  configDirPath,
  accountsConfig: AccountsConfig.load(configDirPath),
  globalConfig: GlobalConfig.load(configDirPath),
  iamConfig: IamConfig.load(configDirPath),
  networkConfig: NetworkConfig.load(configDirPath),
  organizationConfig: OrganizationConfig.load(configDirPath),
  securityConfig: SecurityConfig.load(configDirPath),
  partition: 'aws',
};

/**
 * Build all related stacks
 */
const stacks = new Map<string, NetworkPrepStack>();

for (const region of props.globalConfig.enabledRegions) {
  for (const account of [...props.accountsConfig.mandatoryAccounts, ...props.accountsConfig.workloadAccounts]) {
    const accountId = props.accountsConfig.getAccountId(account.name);

    stacks.set(
      `${account.name}-${region}`,
      new NetworkPrepStack(app, `${AcceleratorStackNames[AcceleratorStage.NETWORK_VPC]}-${accountId}-${region}`, {
        env: {
          account: accountId,
          region,
        },
        ...props,
      }),
    );
  }
}

/**
 * NetworkPrepStack construct test
 */
describe('NetworkPrepStack', () => {
  /**
   * Number of TransitGatewayRouteTable resource test
   */
  test(`${testNamePrefix} TransitGatewayRouteTable resource count test`, () => {
    cdk.assertions.Template.fromStack(stacks.get(`Network-us-east-1`)!).resourceCountIs(
      'AWS::EC2::TransitGatewayRouteTable',
      4,
    );
  });

  // /**
  //  * Number of TransitGateway resource test
  //  */
  // test(`${testNamePrefix} TransitGateway resource count test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).resourceCountIs(
  //     'AWS::EC2::TransitGateway',
  //     1,
  //   );
  // });

  // /**
  //  * Number of RAM ResourceShare resource test
  //  */
  // test(`${testNamePrefix} RAM ResourceShare resource count test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).resourceCountIs(
  //     'AWS::RAM::ResourceShare',
  //     1,
  //   );
  // });

  // /**
  //  * Number of SSM parameter resource test
  //  */
  // test(`${testNamePrefix} SSM parameter resource count test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).resourceCountIs('AWS::SSM::Parameter', 7);
  // });

  // /**
  //  * CoreTransitGatewayRouteTable TransitGatewayRouteTable resource configuration test
  //  */
  // test(`${testNamePrefix} CoreTransitGatewayRouteTable TransitGatewayRouteTable resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       CoreTransitGatewayRouteTableD73AD6A9: {
  //         Type: 'AWS::EC2::TransitGatewayRouteTable',
  //         Properties: {
  //           Tags: [
  //             {
  //               Key: 'Name',
  //               Value: 'core',
  //             },
  //           ],
  //           TransitGatewayId: {
  //             Ref: 'MainTransitGateway66204EF2',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * MainTransitGateway TransitGateway resource configuration test
  //  */
  // test(`${testNamePrefix} MainTransitGateway TransitGateway resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       MainTransitGateway66204EF2: {
  //         Type: 'AWS::EC2::TransitGateway',
  //         Properties: {
  //           AmazonSideAsn: 65521,
  //           AutoAcceptSharedAttachments: 'enable',
  //           DefaultRouteTableAssociation: 'disable',
  //           DefaultRouteTablePropagation: 'disable',
  //           DnsSupport: 'enable',
  //           Tags: [
  //             {
  //               Key: 'Name',
  //               Value: 'Main',
  //             },
  //           ],
  //           VpnEcmpSupport: 'enable',
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * RAM ResourceShare MainTransitGatewayShareMainTransitGatewayShareResourceShare  resource configuration test
  //  */
  // test(`${testNamePrefix} RAM ResourceShare MainTransitGatewayShareMainTransitGatewayShareResourceShare resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       MainTransitGatewayShareResourceShare02109087: {
  //         Type: 'AWS::RAM::ResourceShare',
  //         Properties: {
  //           Name: 'Main_TransitGatewayShare',
  //           Principals: ['Sandbox-arn', '222222222222'],
  //           ResourceArns: [
  //             {
  //               'Fn::Join': [
  //                 '',
  //                 [
  //                   'arn:',
  //                   {
  //                     Ref: 'AWS::Partition',
  //                   },
  //                   ':ec2:us-east-1:111111111111:transit-gateway/',
  //                   {
  //                     Ref: 'MainTransitGateway66204EF2',
  //                   },
  //                 ],
  //               ],
  //             },
  //           ],
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * CoreTransitGatewayRouteTable SegregatedTransitGatewayRouteTable resource configuration test
  //  */
  // test(`${testNamePrefix} CoreTransitGatewayRouteTable SegregatedTransitGatewayRouteTable resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       SegregatedTransitGatewayRouteTableFBA11CE4: {
  //         Type: 'AWS::EC2::TransitGatewayRouteTable',
  //         Properties: {
  //           Tags: [
  //             {
  //               Key: 'Name',
  //               Value: 'segregated',
  //             },
  //           ],
  //           TransitGatewayId: {
  //             Ref: 'MainTransitGateway66204EF2',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * CoreTransitGatewayRouteTable SharedTransitGatewayRouteTable resource configuration test
  //  */
  // test(`${testNamePrefix} CoreTransitGatewayRouteTable SharedTransitGatewayRouteTable resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       SharedTransitGatewayRouteTableDEC04AD4: {
  //         Type: 'AWS::EC2::TransitGatewayRouteTable',
  //         Properties: {
  //           Tags: [
  //             {
  //               Key: 'Name',
  //               Value: 'shared',
  //             },
  //           ],
  //           TransitGatewayId: {
  //             Ref: 'MainTransitGateway66204EF2',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * SSM parameter SsmParamMainTransitGatewayId resource configuration test
  //  */
  // test(`${testNamePrefix} SSM parameter SsmParamMainTransitGatewayId resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       SsmParamMainTransitGatewayId76D30719: {
  //         Type: 'AWS::SSM::Parameter',
  //         Properties: {
  //           Name: '/accelerator/network/transitGateways/Main/id',
  //           Type: 'String',
  //           Value: {
  //             Ref: 'MainTransitGateway66204EF2',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * CoreTransitGatewayRouteTable StandaloneTransitGatewayRouteTable resource configuration test
  //  */
  // test(`${testNamePrefix} CoreTransitGatewayRouteTable StandaloneTransitGatewayRouteTable resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       StandaloneTransitGatewayRouteTableD8B42C98: {
  //         Type: 'AWS::EC2::TransitGatewayRouteTable',
  //         Properties: {
  //           Tags: [
  //             {
  //               Key: 'Name',
  //               Value: 'standalone',
  //             },
  //           ],
  //           TransitGatewayId: {
  //             Ref: 'MainTransitGateway66204EF2',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * SSM parameter SsmParamMaincoreTransitGatewayRouteTableId resource configuration test
  //  */
  // test(`${testNamePrefix} SSM parameter SsmParamMaincoreTransitGatewayRouteTableId resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       SsmParamMaincoreTransitGatewayRouteTableIdC4F7B376: {
  //         Type: 'AWS::SSM::Parameter',
  //         Properties: {
  //           Name: '/accelerator/network/transitGateways/Main/routeTables/core/id',
  //           Type: 'String',
  //           Value: {
  //             Ref: 'CoreTransitGatewayRouteTableD73AD6A9',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * SSM parameter SsmParamMainsegregatedTransitGatewayRouteTableId resource configuration test
  //  */
  // test(`${testNamePrefix} SSM parameter SsmParamMainsegregatedTransitGatewayRouteTableId resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       SsmParamMainsegregatedTransitGatewayRouteTableId8DCAFE8D: {
  //         Type: 'AWS::SSM::Parameter',
  //         Properties: {
  //           Name: '/accelerator/network/transitGateways/Main/routeTables/segregated/id',
  //           Type: 'String',
  //           Value: {
  //             Ref: 'SegregatedTransitGatewayRouteTableFBA11CE4',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * SSM parameter SsmParamMainsharedTransitGatewayRouteTableId resource configuration test
  //  */
  // test(`${testNamePrefix} SSM parameter SsmParamMainsharedTransitGatewayRouteTableId resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       SsmParamMainsharedTransitGatewayRouteTableId2B981DF1: {
  //         Type: 'AWS::SSM::Parameter',
  //         Properties: {
  //           Name: '/accelerator/network/transitGateways/Main/routeTables/shared/id',
  //           Type: 'String',
  //           Value: {
  //             Ref: 'SharedTransitGatewayRouteTableDEC04AD4',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * SSM parameter SsmParamMainstandaloneTransitGatewayRouteTableId resource configuration test
  //  */
  // test(`${testNamePrefix} SSM parameter SsmParamMainstandaloneTransitGatewayRouteTableId resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       SsmParamMainstandaloneTransitGatewayRouteTableIdE6B97388: {
  //         Type: 'AWS::SSM::Parameter',
  //         Properties: {
  //           Name: '/accelerator/network/transitGateways/Main/routeTables/standalone/id',
  //           Type: 'String',
  //           Value: {
  //             Ref: 'StandaloneTransitGatewayRouteTableD8B42C98',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });

  // /**
  //  * SSM parameter SsmParamMainTransitGatewayId resource configuration test
  //  */
  // test(`${testNamePrefix} SSM parameter SsmParamStackId resource configuration test`, () => {
  //   cdk.assertions.Template.fromStack(stacks.get(`Management-us-east-1`)!).templateMatches({
  //     Resources: {
  //       SsmParamStackId521A78D3: {
  //         Type: 'AWS::SSM::Parameter',
  //         Properties: {
  //           Name: '/accelerator/AWSAccelerator-NetworkPrepStack-111111111111-us-east-1/stack-id',
  //           Type: 'String',
  //           Value: {
  //             Ref: 'AWS::StackId',
  //           },
  //         },
  //       },
  //     },
  //   });
  // });
});
