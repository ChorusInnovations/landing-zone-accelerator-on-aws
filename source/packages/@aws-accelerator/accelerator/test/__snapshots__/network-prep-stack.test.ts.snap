// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkPrepStack Construct(NetworkPrepStack):  Snapshot Test 1`] = `
Object {
  "Resources": Object {
    "AcceleratorKeyLookup0C18DA36": Object {
      "DeletionPolicy": "Delete",
      "DependsOn": Array [
        "CustomSsmGetParameterValueCustomResourceProviderLogGroup780D220D",
      ],
      "Properties": Object {
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "CustomSsmGetParameterValueCustomResourceProviderHandlerAAD0E7EE",
            "Arn",
          ],
        },
        "assumeRoleArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":iam::222222222222:role/AWSAccelerator-CrossAccount-SsmParameter-Role",
            ],
          ],
        },
        "invokingAccountID": "333333333333",
        "parameterAccountID": "222222222222",
        "parameterName": "/accelerator/kms/key-arn",
        "region": "us-east-1",
      },
      "Type": "Custom::SsmGetParameterValue",
      "UpdateReplacePolicy": "Delete",
    },
    "CoreTransitGatewayRouteTableD73AD6A9": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "core",
          },
        ],
        "TransitGatewayId": Object {
          "Ref": "MainTransitGateway66204EF2",
        },
      },
      "Type": "AWS::EC2::TransitGatewayRouteTable",
    },
    "CustomSsmGetParameterValueCustomResourceProviderHandlerAAD0E7EE": Object {
      "DependsOn": Array [
        "CustomSsmGetParameterValueCustomResourceProviderRoleB3AFDDB2",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": "cdk-hnb659fds-assets-333333333333-us-east-1",
          "S3Key": "1cd3b5004f506f3f38fb48c80c9b452b94c5e1b2f1e54f0ad99a27e8f47cbe49.zip",
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": Object {
          "Fn::GetAtt": Array [
            "CustomSsmGetParameterValueCustomResourceProviderRoleB3AFDDB2",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomSsmGetParameterValueCustomResourceProviderLogGroup780D220D": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "LogGroupName": Object {
          "Fn::Join": Array [
            "",
            Array [
              "/aws/lambda/",
              Object {
                "Ref": "CustomSsmGetParameterValueCustomResourceProviderHandlerAAD0E7EE",
              },
            ],
          ],
        },
        "RetentionInDays": 365,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomSsmGetParameterValueCustomResourceProviderRoleB3AFDDB2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": Array [
                    "ssm:GetParameters",
                    "ssm:GetParameter",
                    "ssm:DescribeParameters",
                  ],
                  "Effect": "Allow",
                  "Resource": Array [
                    "*",
                  ],
                  "Sid": "SsmGetParameterActions",
                },
                Object {
                  "Action": Array [
                    "sts:AssumeRole",
                  ],
                  "Effect": "Allow",
                  "Resource": Array [
                    Object {
                      "Fn::Join": Array [
                        "",
                        Array [
                          "arn:",
                          Object {
                            "Ref": "AWS::Partition",
                          },
                          ":iam::222222222222:role/AWSAccelerator-CrossAccount-SsmParameter-Role",
                        ],
                      ],
                    },
                  ],
                  "Sid": "StsAssumeRoleActions",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "Inline",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "MainTransitGateway66204EF2": Object {
      "Properties": Object {
        "AmazonSideAsn": 65521,
        "AutoAcceptSharedAttachments": "enable",
        "DefaultRouteTableAssociation": "disable",
        "DefaultRouteTablePropagation": "disable",
        "DnsSupport": "enable",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "Main",
          },
        ],
        "VpnEcmpSupport": "enable",
      },
      "Type": "AWS::EC2::TransitGateway",
    },
    "MainTransitGatewayShareResourceShare02109087": Object {
      "Properties": Object {
        "Name": "Main_TransitGatewayShare",
        "Principals": Array [
          "Sandbox-arn",
          "222222222222",
        ],
        "ResourceArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":ec2:us-east-1:333333333333:transit-gateway/",
                Object {
                  "Ref": "MainTransitGateway66204EF2",
                },
              ],
            ],
          },
        ],
      },
      "Type": "AWS::RAM::ResourceShare",
    },
    "SegregatedTransitGatewayRouteTableFBA11CE4": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "segregated",
          },
        ],
        "TransitGatewayId": Object {
          "Ref": "MainTransitGateway66204EF2",
        },
      },
      "Type": "AWS::EC2::TransitGatewayRouteTable",
    },
    "SharedTransitGatewayRouteTableDEC04AD4": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "shared",
          },
        ],
        "TransitGatewayId": Object {
          "Ref": "MainTransitGateway66204EF2",
        },
      },
      "Type": "AWS::EC2::TransitGatewayRouteTable",
    },
    "SsmParamAcceleratorVersionFF83282D": Object {
      "Properties": Object {
        "Name": "/accelerator/AWSAccelerator-NetworkPrepStack-333333333333-us-east-1/version",
        "Type": "String",
        "Value": "0.6.0-rc001",
      },
      "Type": "AWS::SSM::Parameter",
    },
    "SsmParamMainTransitGatewayId76D30719": Object {
      "Properties": Object {
        "Name": "/accelerator/network/transitGateways/Main/id",
        "Type": "String",
        "Value": Object {
          "Ref": "MainTransitGateway66204EF2",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "SsmParamMaincoreTransitGatewayRouteTableIdC4F7B376": Object {
      "Properties": Object {
        "Name": "/accelerator/network/transitGateways/Main/routeTables/core/id",
        "Type": "String",
        "Value": Object {
          "Ref": "CoreTransitGatewayRouteTableD73AD6A9",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "SsmParamMainsegregatedTransitGatewayRouteTableId8DCAFE8D": Object {
      "Properties": Object {
        "Name": "/accelerator/network/transitGateways/Main/routeTables/segregated/id",
        "Type": "String",
        "Value": Object {
          "Ref": "SegregatedTransitGatewayRouteTableFBA11CE4",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "SsmParamMainsharedTransitGatewayRouteTableId2B981DF1": Object {
      "Properties": Object {
        "Name": "/accelerator/network/transitGateways/Main/routeTables/shared/id",
        "Type": "String",
        "Value": Object {
          "Ref": "SharedTransitGatewayRouteTableDEC04AD4",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "SsmParamMainstandaloneTransitGatewayRouteTableIdE6B97388": Object {
      "Properties": Object {
        "Name": "/accelerator/network/transitGateways/Main/routeTables/standalone/id",
        "Type": "String",
        "Value": Object {
          "Ref": "StandaloneTransitGatewayRouteTableD8B42C98",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "SsmParamStackId521A78D3": Object {
      "Properties": Object {
        "Name": "/accelerator/AWSAccelerator-NetworkPrepStack-333333333333-us-east-1/stack-id",
        "Type": "String",
        "Value": Object {
          "Ref": "AWS::StackId",
        },
      },
      "Type": "AWS::SSM::Parameter",
    },
    "StandaloneTransitGatewayRouteTableD8B42C98": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "standalone",
          },
        ],
        "TransitGatewayId": Object {
          "Ref": "MainTransitGateway66204EF2",
        },
      },
      "Type": "AWS::EC2::TransitGatewayRouteTable",
    },
  },
}
`;
